temporalio/ui
# add new combobox - remove old namespace switcher components

## Description & motivation 💭 
Adds new `<Combobox />` and `<ComboboxOption />` components, and use them in the namespace switcher.

Combobox is designed to be flexible such that it can take an array of strings and render its own ComboboxOptions, or it can take an array of any object type, and exposes each option in a slot. It also has keyboard navigation and proper aria labels baked in for screen reader and keyboard navigation.

Also refactor`<MenuContainer />`, `<MenuButton />`, `<Menu />` and `<MenuItem />` to have keyboard navigation, hover, and focus states baked in, and use these components anywhere a dropdown menu is utilized. This includes our existing `<Select />` component.

Also remove `src/lib/components/dropdown-menu.svelte`, `src/lib/holocene/dropdown-menu.svelte`, `src/lib/holocene/autocomplete.svelte`, and `src/lib/holocene/dropdown-button/`. 

Move Menu components out of `src/lib/holocene/primitives` to `src/lib/holocene/menu`. 

### Screenshots (if applicable) 📸 

https://github.com/temporalio/ui/assets/11775628/f0379724-15d0-4cbe-903c-0b12687542f5

### Design Considerations 🎨 
- Need to confirm with Ash that the Component matches Figma
## Testing 🧪 

### How was this tested 👻 


### Steps for others to test: 🚶🏽‍♂️🚶🏽‍♀️ 
Either in the Web UI or the Holocene preview URL, verify the following
- tabbing into the Combobox displays a focus state and places a cursor in the input
  - hitting "Enter" or "DownArrow" opens the Combobox options and focuses the first option
  - typing opens the Combobox options
  - `Tab` and `Shift+Tab` cycle through the input and each option, without trapping focus
  - `UpArrow` and `DownArrow` cycle through the input an each option _with_ trapping focus
  - `Enter` and `Space` select a Combobox Option while it is focused
- clicking on the Combobox displays a focus state and opens the Combobox options
  - `Tab` and `Shift+Tab` cycle through the input and each option, without trapping focus
  - `UpArrow` and `DownArrow` cycle through the input an each option _with_ trapping focus
  - `Enter` and `Space` select a Combobox Option while it is focused
- A checkmark is displayed for the selected option
- Disabled options cannot be selected

## Checklists

### Draft Checklist 

### Merge Checklist 

### Issue(s) closed 

## Docs

### Any docs updates needed?
